File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) 
    {   
        // bool dupe = false; 
        // for(int i = 0 ; i < nums.size() ; i++)
        // {
        //     for(int j = i+1 ; j<nums.size() ; j++)
        //     {
        //         if(nums[i] == nums[j])
        //         dupe = true;
        //     }
        // }
        set <int> dup ; 
        for(int i = 0 ; i < nums.size() ; i++)
        {
            dup.insert(nums[i]);
        }
        if(dup.size() < nums.size())
        return true; 
        else 
        return false; 

    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        freq = {}
        for i in nums:
            if i in freq :
                freq[i] = freq[i] + 1 
            else :
                freq[i] = 1
        l  = []
        for i in range(len(nums)):
            l.append(1)
        print(l)
        if freq.values() == l :
            return False 
        else :
            return True 
